test/e2e_env/universal/gateway/gateway.go

```diff
@@ -0,0 +1,267 @@
+		It("shouldn't error with gateway-proxy import", func() {
+			Expect(
+				env.Cluster.GetKumactlOptions().KumactlApplyFromString(fmt.Sprintf(`
+type: ProxyTemplate
+mesh: %s
+name: edge-gateway
+selectors:
+  - match:
+      kuma.io/service: edge-gateway
+conf:
+  imports:
+    - gateway-proxy
+  modifications:
+    - cluster:
+        operation: add
+        value: |
+          name: test-cluster
+          connectTimeout: 5s
+          type: STATIC
+`, mesh)),
+			).To(Succeed())
+		})
+	})
+
+	Context("when a rate limit is configured", func() {
+		BeforeAll(func() {
+			Expect(
+				env.Cluster.GetKumactlOptions().KumactlApplyFromString(fmt.Sprintf(`
+type: RateLimit
+mesh: %s
+name: echo-rate-limit
+sources:
+- match:
+    kuma.io/service: edge-gateway
+destinations:
+- match:
+    kuma.io/service: echo-service
+conf:
+  http:
+    requests: 5
+    interval: 10s
+`, mesh)),
+			).To(Succeed())
+		})
+
+		It("should be rate limited", func() {
+			proxyRequestsWithRateLimit(env.Cluster,
+				GatewayAddressPort("gateway-proxy", gatewayPort))
+		})
+	})
+
+	Context("when targeting a HTTPS gateway", func() {
+		BeforeAll(func() {
+			// Delete the default gateway that the test fixtures create.
+			Expect(
+				env.Cluster.GetKumactlOptions().KumactlDelete("meshgateway", "edge-gateway", mesh),
+			).To(Succeed())
+
+			// And replace it with a HTTPS gateway.
+			Expect(
+				env.Cluster.GetKumactlOptions().KumactlApplyFromString(fmt.Sprintf(`
+type: MeshGateway
+mesh: %s
+name: edge-https-gateway
+selectors:
+- match:
+    kuma.io/service: edge-gateway
+conf:
+  listeners:
+  - port: 8080
+    protocol: HTTPS
+    hostname: example.kuma.io
+    tls:
+      mode: TERMINATE
+      certificates:
+      - secret: example-kuma-io-certificate
+    tags:
+      hostname: example.kuma.io
+  - port: 8081
+    protocol: HTTPS
+    tls:
+      mode: TERMINATE
+      certificates:
+      - secret: example-kuma-io-certificate
+`, mesh)),
+			).To(Succeed())
+
+			cert, key, err := CreateCertsFor("example.kuma.io")
+			Expect(err).To(Succeed())
+
+			payload := base64.StdEncoding.EncodeToString([]byte(strings.Join([]string{key, cert}, "\n")))
+
+			// Create the TLS secret containing the self-signed certificate and corresponding private key.
+			Expect(
+				env.Cluster.GetKumactlOptions().KumactlApplyFromString(fmt.Sprintf(`
+type: Secret
+mesh: %s
+name: example-kuma-io-certificate
+data: %s
+	`, mesh, payload)),
+			).To(Succeed())
+
+			Expect(
+				env.Cluster.GetKumactlOptions().KumactlList("meshgateways", mesh),
+			).To(ContainElement("edge-https-gateway"))
+		})
+
+		It("should proxy simple HTTPS requests", func() {
+			addr := net.JoinHostPort("example.kuma.io", strconv.Itoa(8080))
+			proxySecureRequests(
+				env.Cluster,
+				"universal",
+				addr,
+				client.Resolve(addr, GatewayAddress("gateway-proxy")),
+			)
+
+			proxySecureRequests(
+				env.Cluster,
+				"universal",
+				GatewayAddressPort("gateway-proxy", 8081),
+			)
+		})
```

Comment:
I think we have 2 test cases here
```
1) should proxy https requests
1) should proxy https requests via hostname
```
also I'd use DescribeTabe and inline `proxySecureRequests`, but that's my opinion

Reply:
I think the function is a bit more readable as is than converting the different calls into a `testCase struct` or interpreting arguments to the `DescribeTable` test function.